packages/generate-arg-types/src/types.ts

```diff
@@ -0,0 +1,143 @@
+import { z } from "zod";
+
+// The names of "control" here are trying to match the names of the controls in Storybook
+// https://storybook.js.org/docs/react/essentials/controls
+
+const common = {
+  required: z.boolean(),
+  label: z.string().optional(),
+};
+
+const Number = z.object({
+  control: z.literal("number"),
+  dataType: z.literal("number"),
+  defaultValue: z.number().nullable(),
+  ...common,
+});
+
+const Range = z.object({
+  control: z.literal("range"),
+  dataType: z.literal("number"),
+  defaultValue: z.number().nullable(),
+  ...common,
+});
+
+const Text = z.object({
+  control: z.literal("text"),
+  dataType: z.string(),
+  defaultValue: z.string().nullable(),
+  ...common,
+});
+
+const MultilineText = z.object({
+  control: z.literal("multiline-text"),
```

Comment:
add a comment that this has been discussed with storybook team
https://discord.com/channels/@me/1005106717870260275/1075352596027551744
and resulted in https://github.com/storybookjs/storybook/issues/21100